ovos_dinkum_listener/service.py

```diff
@@ -161,6 +161,24 @@ def __init__(self, on_ready=on_ready, on_error=on_error,
         listener = self.config["listener"]
         self.voice_loop = self._init_voice_loop(listener)
 
+    def _validate_message_context(self, message, native_sources=None):
+        """ used to determine if a message should be processed or ignored
+        only native sources should trigger on mycroft.mic.listen
+        """
+        if not message or not self.validate_source:
```

Comment:
What is the reasoning to make `validate_source` configurable?
I feel we should be strict. Below (`handle_listen`) would also handle None messages (for whatever reason) and in case of a deactivated context check

Reply:
it is to allow subclassing, eg https://github.com/JarbasHiveMind/HiveMind-voice-sat/pull/38

a hivemind client has the equivalent to a dynamic "native source", it wants to skip this check always, if the master sent it, it is to be executed unconditionally

Reply:
In my mind it is a a combination of strict check of native source and placement. This is not really a context check, but a bypass.

Reply:
a bypass is exactly the goal, to allow any subclass to disable this check completely, but the bypass never happens under OVOS, it is only when using this code as a python library  (afaik currently only hivemind voice sat does this)

under hivemind there is no concept of native source, there is only commands sent from master that should never be skipped, bypassing the check is the intended behavior. under the hivemind trust model any check (for any message) would happen in hivemind-core and just not be sent to the satellites, from a sat POV you got the message you handle it. "native source" does not apply to a satellite, it's all just an order you must obey, "native" events are never generated in the sat itself. theres only requests (sat -> master) and commands (master -> sat)